compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -969,6 +969,9 @@ impl<'a> PartialEvaluator<'a> {
                 expr_package_span,
             )),
             ExprKind::Return(expr_id) => self.eval_expr_return(*expr_id),
+            ExprKind::Struct(..) => {
```

Comment:
This is confusing to me, as because the AssignField and the UpdateField are considered invalid, but it seems like the same reasoning could be applied to them.

Reply:
Ah, you are right, those are runtime failures. Then I think updating this to be an `Err` rather than a panic would make it consistent with those. That way it doesn't crash but fails gracefully and produces a diagnostic in editor.

Reply:
I've changed it to be an `Err` like the others, instead of a panic.